app/assets/javascripts/discourse/app/components/sidebar/categories-section.js

```diff
@@ -20,11 +26,28 @@ export default class SidebarCategoriesSection extends GlimmerComponent {
 
   @cached
   get sectionLinks() {
-    return this.site.trackedCategoriesList.map((trackedCategory) => {
-      return new CategorySectionLink({
-        category: trackedCategory,
-        topicTrackingState: this.topicTrackingState,
-      });
+    const links = [];
+
+    for (const category of this.currentUser.sidebarCategories) {
+      links.push(
+        new CategorySectionLink({
+          category,
+          topicTrackingState: this.topicTrackingState,
+        })
+      );
+    }
```

Comment:
I'm intentionally micro optimising here since the sidebar is going to be rendered frequently and benchmarks show that for loops are generally faster. There is little impact on readability here as well so I think the `for` loop is OK.